### PR TITLE
react native autosuggest

### PR DESCRIPTION
**Pull request by maksim-v**

### CODE REVIEW

File: src/components/Autosuggest/Autosuggest.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      setCollection(
+        items.filter(
+          item => item.value.toLowerCase().indexOf(text.toLowerCase()) !== -1
+        )
+      );
+    },
+    [collection, onChangeText, value]
+  );
+
+  const handleFocus = useCallback(() => setOpen(true), [isOpened]);
```
Comment (elliolir) [inline src/components/Autosuggest/Autosuggest.js:58]: do we really need to optimize this simple line with `useCallback `? 🙂

File: src/components/Autosuggest/Autosuggest.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      <TextInput
+        style={styles.input}
+        onFocus={handleFocus}
+        ref={inputRef}
+        onChangeText={onChangeValue}
+        onSubmitEditing={hideList}
+        value={value}
+        {...props}
+      />
+      {isOpened && (
```
Comment (elliolir) [inline src/components/Autosuggest/Autosuggest.js:88]: can we display list only when user typed at least three characters?

Comment (AlexeyKaryka) [reply src/components/Autosuggest/Autosuggest.js:88]: As far as I imagine flow of this component, above case shouldn't be applied.
When user touches input, dropdown occurs, then user starts typing desired value, for instance bird name "Чиж", after first typed letter dropdown should filter all values that don't start from "ч/Ч"

File: src/components/Autosuggest/Autosuggest.js (hunk 1 above)
Comment (maksim-v) [reply src/components/Autosuggest/Autosuggest.js:58]: @DimatiBart yes, without `useCallback` these functions will be recreated on every render, on every click 🙂

File: src/components/Autosuggest/Autosuggest.js (hunk 2 above)
Comment (maksim-v) [reply src/components/Autosuggest/Autosuggest.js:88]: @AlexeyKaryka yes, you are right 🙂

**Pull request merged**